react-events/my-app/src/App.tsx

```diff
@@ -0,0 +1,55 @@
+import { Header } from './Header';
+import { Image } from './Image';
+import { Caption } from './Caption';
+import { Description } from './Description';
+import { Button } from './Button';
+import { useState } from 'react';
+import './App.css';
+
+const srcs = ['/starry-sky.jpeg', '/cool-kitty.jpg', '/cool-car.jpg'];
+const captions = ['A Beautiful Image of Space', 'A Cool Cat', 'A Cool Car'];
+const descriptions = [
+  'This is a wonderful sky full of stars!',
+  'This is a picture of a very cool cat!',
+  'This is a picture of a very cool car!',
+];
+
+export function App() {
+  const [index, setIndex] = useState(0);
+  function handleCaptionClick() {
+    if (index >= captions.length - 1) {
+      setIndex(0);
+    } else {
+      setIndex(index + 1);
+    }
+  }
+  function handleDescriptionClick() {
+    if (index >= descriptions.length - 1) {
+      setIndex(0);
+    } else {
+      setIndex(index + 1);
+    }
+  }
+
+  function handleImageclick() {
+    if (index >= srcs.length - 1) {
+      setIndex(0);
+    } else {
+      setIndex(index + 1);
+    }
+  }
+  function handleNextClick() {
```

Comment:
This is a lot closer, but you don't need 3 click handler functions, just one. You are using `index` to control the display in all 3 components, so you only need to update it once.

Reply:
Good! Now this function doesn't really have much use since it simply calls handleClick. You could replace handleNextClick on line 35 with handleClick, and then delete this function.